### DISCUSSION TITLE
Can loginByToken API use SAML-token/session-id to get session to vcenter using govmomi

### DISCUSSION THREAD

**Discussion by jainrajiv02** [vmware/govmomi#3110] [Community Help] [answered]
I am calling below API to get session token 
https://developer.vmware.com/apis/vsphere-automation/latest/cis/api/session/post/

as per the documentation this token can be used for further calls to loginByToken() or cloneSession() APIs to get sessions

is this the recommended way to connect with the vSphere MOB APIs. Please revert in case of other recommendations? we wanted to get it done using govmomi.

Any sample approach would be appreciated.

**Comment by jainrajiv02** [comment]
Hello All,

I am able to create session using below code

vCenterURL := "https://administrator@vsphere.local:Netapp1!@10.224.57.70/sdk"
.
.
client, err := govmomi.NewClient(ctx, u, true)
.
.

Now for subsequent calls, I dont want to use cedentials again, can we use something like session-id ?

**Reply by dougm** [reply, marked as answer]
Hi, you won't be using credentials again with `client` in that example. After the initial Login(), an http cookie is set and used to authenticate subsequent calls. It is transparent within the Go http.Client underneath, but if you need to use the session ID with another client:
```go
		cookies := client.Jar.Cookies(client.URL())
		for _, cookie := range cookies {
			if cookie.Name == soap.SessionCookieName {
				fmt.Println(cookie.Value)
			}
		}
```
